src/semeio/forward_models/__init__.py

```diff
@@ -54,7 +57,13 @@ def validate_pre_realization_run(
         return fm_step_json
 
     def validate_pre_experiment(self, fm_step_json: ForwardModelStepJSON) -> None:
-        pass
+        errors = DesignKWScript.validate_configuration(fm_step_json.get("argList")[0])
```

Comment:
well, this still does not make sense. Ie. when `should_fail_on_validation_error` we don't need to do any validation based on the code at least. Then it can all go into:
```python
if should_fail_on_validation_error:
      DesignKWScript.validate_configuration(fm_step_json.get("argList")[0])
      ...
```

Reply:
I agree, but I do not know if I want to call a `ForwardModelStepWarning` method directly from  DesignKWScript. The call to `ForwardModelStepWarning.warn(...)` only makes sense when called from a `ForwardModelStepPlugin` context.

Reply:
Should definily only validate inside the if statement as there are at least two bugs that causes validation to crash, see https://github.com/equinor/semeio/issues/641. Should probably also fix those bugs and have a defensive try-catch around the validation.

Reply:
We do not get the argument key, only the value from `ForwardModelStepJSON`. I think it is better to assume the final argument value is for validation instead of looping over and looking for a string of a boolean
```
fm_step_json={'name': 'DESIGN_KW', 'executable': 'design_kw', 'target_file': None, 'error_file': None, 'start_file': None, 'stdout': 'DESIGN_KW.stdout.0', 'stderr': 'DESIGN_KW.stderr.0', 'stdin': None, 'argList': ['template.yml.tmpl', '<result_file>', 'true'], 'environment': {'_ERT_ITERATION_NUMBER': '0', '_ERT_REALIZATION_NUMBER': '0', '_ERT_RUNPATH': '/Users/JONAK/Documents/FMU/SCOUT/ert/test-data/poly_example/poly_out/realization-0/iter-0'}, 'exec_env': None, 'max_running_minutes': None}
```